website/docs/advanced-topics/immutable.mdx

```diff
@@ -0,0 +1,24 @@
+---
+title: 'Immutable Types'
+description: 'Immutable data structures for Yew'
+---
+
+## What are immutable types?
+
+These are types that you can instantiate but never mutate the values. In order
+to update a value, you must instantiate a new value.
+
+## Why using immutable types?
+
+In a React-like framework, properties are propagated from ancestors to
```

Comment:
```suggestion
Properties are propagated from ancestors to
```

Reply:
Fine by me but I'm afraid that if we try to distance ourselves too much from "being a clone of react" we will lose the understanding.

Properties are propagated from ancestors to children BECAUSE it's how these kind of frameworks work. On the other hand, the "word" property can have a completely different meaning in other frameworks. For instance, in VisualBASIC (I haven't work on GUI for a long time lol), "properties" meant the thingies you set on the objects and activate/deactivate a specific feature. There is no notion of propagation at all.

I'm also personally convinced that people who start with Yew are looking for React in Rust. The documentation should help them navigate through it.

Reply:
> I'm also personally convinced that people who start with Yew are looking for React in Rust.

It was quite the opposite for me. I learnt Yew before I learnt React. I believe this should be better a way to word this:
```suggestion
Properties, like in React, are propagated from ancestors to
```

I'm kind of hesitant of documentation making Yew sound like it's a React clone or (more importantly) making React be the be-all and end-all of web frameworks/libraries. It is inspired by React but there are certain areas where we shouldn't (or just can't) following what React does because Rust is a entirely a different language and requires a different mindset to work with. It's not the case here though.

Reply:
> It is inspired by React but there are certain areas where we shouldn't (or just can't) following what React does because Rust is a entirely a different language and requires a different mindset to work with.

:100: I totally agree with you. Definitely not the same languages and problems are often solved very differently. I'm also discussing about this here too: https://github.com/yewstack/yew/issues/1843#issuecomment-1157240010

> I'm kind of hesitant of documentation making Yew sound like it's a React clone or (more importantly) making React be the be-all and end-all of web frameworks/libraries

I think you misunderstood me. I never said React is the be-all and end-all of anything.

My point is that React is one huge piece of influence in web development for years and people arrive in Rust and in Yew with already years of experiences in React. It seems it is not your case (good for you actually) but for the rest of us we need to unlearn some things and adapt to the Rust reality.

Reply:
(I kept the conversation "unresolved" so it is not hidden in the PR but I already committed your proposal with the ", like in React,")